Mathlib/Data/Set/Sups.lean

```diff
@@ -170,6 +169,18 @@ theorem sups_inter_subset_right : s ⊻ (t₁ ∩ t₂) ⊆ s ⊻ t₁ ∩ s ⊻
   image2_inter_subset_right
 #align set.sups_inter_subset_right Set.sups_inter_subset_right
 
+lemma image_sups (f : F) (s t : Set α) : f '' (s ⊻ t) = f '' s ⊻ f '' t :=
+  image_image2_distrib $ map_sup f
+
+lemma subset_sups_self : s ⊆ s ⊻ s := λ _a ha ↦ mem_sups.2 ⟨_, ha, _, ha, sup_idem⟩
+lemma sups_self_subset : s ⊻ s ⊆ s ↔ SupClosed s := sups_subset_iff
+
+@[simp] lemma sups_self : s ⊻ s = s ↔ SupClosed s :=
+  subset_sups_self.le.le_iff_eq.symm.trans sups_self_subset
+
+lemma sep_sups_le (s t : Set α) (a : α) :
+    {b ∈ s ⊻ t | b ≤ a} = {b ∈ s | b ≤ a} ⊻ {b ∈ t | b ≤ a} := by ext; aesop
```

Comment:
Would it make sense to state this as `(s ⊻ t) \inter Iic a = ...`? Or is that less useful to you?

Reply:
For the `Finset` version, I must use `filter` since my order isn't locally finite. For the `Set` version, I don't really mind, except that using `sep` is more consistent than intersecting with `Iic a`.

What do you think? (I don't think it's too important so I'll merge the PR as is, but I'll modify that file quite a lot in the next few weeks so I'm happy to change later)